### PR TITLE
Fix country region to be device local region

### PR DESCRIPTION
**Pull request by ehab-al-cko**
## Proposed changes

When `iso3166Alpha2` is not set by the merchant then use local device region 

## Types of changes

What types of changes does your code introduce to Frames Ios?
_Put an `x` in the boxes that apply_

* [x] Bugfix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [ ] Lint and unit tests pass locally with my changes
* [x] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: Source/UI/NewUI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift [hunk 1]
```diff
@@ -18,8 +18,8 @@ final class DefaultBillingFormViewModel: BillingFormViewModel {
     var errorFlagOfCellType = [Int: Bool]()
     var textValueOfCellType = [Int: String]()
 
-    private var country : Country?
     private var countryRow: Int?
+    private(set) var country : Country?
```
Comment (alex-ioja-yang-cko) [inline Source/UI/NewUI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift:22]: Why the access control change?

File: Source/UI/NewUI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift [hunk 2]
```diff
@@ -74,6 +74,9 @@ final class DefaultBillingFormViewModel: BillingFormViewModel {
             textValueOfCellType[type.index] = hasValue ? value : (!isMandatory ? "" : nil)
             errorFlagOfCellType[type.index] = hasValue ? false : isMandatory
         }
+        if let country = country, country.iso3166Alpha2.isEmpty {
+            self.country = Country(iso3166Alpha2: Locale.current.regionCode ?? "", dialingCode: nil)
```
Comment (alex-ioja-yang-cko) [inline Source/UI/NewUI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift:78]: feels like a complicated way of doing it. Couldn't we just target the initialiser directly?

```
    public init(iso3166Alpha2: String, dialingCode: String?) {
        self.iso3166Alpha2 = iso3166Alpha2.isEmpty ? Locale.current.regionCode : iso3166Alpha2
        self.dialingCode = dialingCode
    }
```

We can then just test this simply with Country initialiser unit test.

Comment (alex-ioja-yang-cko) [reply Source/UI/NewUI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift:78]: The reason I say this feels complicated is that it feels a little hacky, after setup to check if something exists, then if it exists but lacks something, to override.

File: Source/UI/NewUI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift (hunk 1 above)
Comment (ehab-al-cko) [reply Source/UI/NewUI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift:22]: For unit testing

File: Source/UI/NewUI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift (hunk 2 above)
Comment (deepesh-vasthimal-cko) [reply Source/UI/NewUI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift:78]: Ok makes sense for now we should ideally be making these changes in checkout iOS SDK if needed. but for now I am fine with this.

**Review by deepesh-vasthimal-cko — APPROVED** [review]

**Review by alex-ioja-yang-cko — APPROVED** [review]

**Pull request merged**